flexget/plugins/search_t411.py

```diff
@@ -0,0 +1,275 @@
+
+from flexget.manager import Session
+from flexget.db_schema import versioned_base
+
+from requests.auth import AuthBase
+from requests.compat import quote_plus
+
+from datetime import datetime, timedelta
+
+from sqlalchemy import Column, Unicode, Integer, DateTime
+
+log = logging.getLogger('t411')
+Base = versioned_base('t411', 0)
+
+__author__ = 'blAStcodeM & gregaou'
+
+T411_BASE_URL = "http://api.t411.in"
+
+CATEGORIES = {
+    'Animation': 455,
+    'Animation-Serie': 637,
+    'Concert': 633,
+    'Documentaire': 634,
+    'Emission-TV': 639,
+    'Film': 631,
+    'Serie-TV': 433,
+    'Spectacle': 635,
+    'Sport': 636,
+    'Video-clips': 402
+}
+
+SUB_CATEGORIES = {
+    'Anglais': [17, 540],
+    'VFF': [17, 541],
+    'Muet': [17, 722],
+    'Multi-Francais': [17, 542],
+    'Multi-Quebecois': [17, 1160],
+    'VFQ': [17, 719],
+    'VFSTFR': [17, 720],
+    'VOASTA': 0,
+    'VOSTFR': [17, 721],
+
+    'BDrip-BRrip-SD': [7, 8],
+    'BDrip-SD': [7, 8],
+    'Bluray-4K': [7, 1171],
+    'Bluray-Full-Remux': [7, 17],
+    'BRrip-SD': [7, 8],
+    'DVD-R-5': [7, 13],
+    'DVD-R-9': [7, 14],
+    'DVDrip': [7, 10],
+    'HDlight-1080p': [7, 1208],
+    'HDlight-720p': [7, 1218],
+    'HDrip-1080p': [7, 16],
+    'HDrip-720p': [7, 15],
+    'TVrip-SD': [7, 11],
+    'TVripHD-1080p': [7, 1162],
+    'TVripHD-720p': [7, 12],
+    'VCD-SVCD-VHSrip': [7, 18],
+    'WEBrip': [7, 19],
+    'WEBripHD-1080p': [7, 1174],
+    'WEBripHD-720p': [7, 1175],
+    'WEBripHD-4K': [7, 1182],
+
+    '2D': [9, 22],
+    '3D-Converti-Amateur': [9, 1045],
+    '3D-Converti-Pro': [9, 24],
+    '3D-Natif': [9, 23],
+}
+
+SUB_CATEGORIES_SERIES = {
+    'Anglais': [51, 1209],
+    'VFF': [51, 1210],
+    'Muet': [51, 1211],
+    'Multi-Francais': [51, 1212],
+    'Multi-Quebecois': [51, 1213],
+    'VFQ': [51, 1214],
+    'VFSTFR': [51, 1215],
+    'VOASTA': [51, 1217],
+    'VOSTFR': [51, 1216],
+}
+
+
+class t411Account(Base):
+    __tablename__ = 't411_account'
+    id = Column(Integer, primary_key=True, autoincrement=True, nullable=False)
+    username = Column(Unicode, index=True)
+    token = Column(Unicode)
+    expiry_time = Column(DateTime)
+
+
+class t411Auth(AuthBase):
+
+    """ Attaches HTTP Token Authentication to the given Request object."""
+
+    def get_auth(self):
+        url_auth = T411_BASE_URL + "/auth"
+        log.debug("Getting token from : %s ", url_auth)
+        params = {'username': self.username, 'password': self.password}
+        res = requests.post(url_auth, data=params).json()
+        if 'error' in res.keys():
+            raise plugin.PluginError(res['error'])
+
+        if 'token' not in res.keys():
+            raise plugin.PluginError("Unable to get token ! (%s)" % res)
+
+        return res['token']
+
+    def update_token(self, db_session, token):
+        db_session.token = token
+        db_session.expiry_time = datetime.now() + timedelta(days=90)
+
+    def get_token(self):
+        db_session = Session()
```

Comment:
If you use `with Session() as db_session:`, then you don't have to call commit().